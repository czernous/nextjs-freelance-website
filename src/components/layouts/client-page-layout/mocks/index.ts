import { IClientPageLayoutProps } from '../../../../interfaces';

export const clientPageMock: IClientPageLayoutProps = {
  pageTitle: 'About',
  meta: {
    metaDescription:
      'This is a dummy description designed for testing purposes',
    metaKeywords: 'some meta keywords,testing,typescript,blog',
    openGraph: {
      title: 'This title will be shown on social media',
      description: 'This description will be shown on social media',
      imageUrl: '',
      type: '',
      url: '',
    },
  },
};
