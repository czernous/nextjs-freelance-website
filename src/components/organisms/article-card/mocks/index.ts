import { IArticleCard, IImage } from '../../../../interfaces';
import { base64ImgSmall, base64ImgMedium, base64ImgLarge } from './images.mock';

const imageMock: IImage = {
  publicId: 'fakeurl.com/asd',
  version: 1,
  signature: 'fakeSignature',
  width: 1000,
  height: 666,
  format: 'jpg',
  resourceType: 'fake resource',
  bytes: 111,
  type: '',
  url: base64ImgLarge,
  secureUrl: base64ImgLarge,
  responsiveUrls: [
    { width: 1000, url: base64ImgLarge },
    { width: 700, url: base64ImgMedium },
    { width: 500, url: base64ImgSmall },
  ],
  thumbnailUrl: base64ImgSmall,
  blurredImageUrl: base64ImgSmall,
  path: 'fakeurl',
  id: {
    timestamp: 111,
    machine: 111,
    pid: -999,
    increment: 111,
    creationTime: '2022-06-25T15:44:40.411Z',
  },
  _id: '12345',
  createdAt: '2022-06-25T15:44:40.411Z',
  updatedAt: '2022-06-25T15:44:40.411Z',
  altText: 'mock image',
};

export const cardMock: IArticleCard = {
  title: 'Article title',
  unoptimized: true,
  description:
    'Some random article short summary. This will be taken from the article itself or prefilled when the article is posted',
  ctaText: 'Read more',
  ctaUrl: 'https//google.com',
  image: imageMock,
};
