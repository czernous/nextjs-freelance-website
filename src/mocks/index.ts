export * from './page-data';
export * from './posts';
export * from './paginated-posts';
