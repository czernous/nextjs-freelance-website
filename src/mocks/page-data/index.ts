import { IHomePage, IServicesPage } from '@src/interfaces';

export const homePageMock: IHomePage = {
  pageFields: {
    ctaBtnHref: 'https://google.com',
    ctaBtnText: 'Get stuff',
    ctaHeadline: 'Are you looking to get awesome stuff??',
    ctaSubheadline: 'We have the most awesome stuff on the internet!',
  },
  slug: 'home',
  meta: {
    metaDescription: 'Test',
    metaKeywords: 'test',
    openGraph: {
      title: 'Test',
      description: 'test',
      imageUrl: 'http://g.com',
    },
  },

  updatedAt: '2023-02-16T15:47:25.559Z',
};

export const servicesPageMock: IServicesPage = {
  pageFields: {
    content:
      '<h2>Service #1</h2><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis nunc libero. Nullam tincidunt lectus ut maximus dignissim. Mauris et risus in metus ornare convallis. Nullam in orci leo. Aenean sit amet arcu vitae ipsum mollis blandit. Pellentesque pellentesque sapien mollis, pharetra magna sit amet, varius eros. Donec eu dui ullamcorper ligula rutrum tincidunt. Sed eu venenatis mi. Nullam rutrum sagittis tortor, id feugiat dolor commodo a. Nullam venenatis rutrum elementum. Vestibulum vehicula lectus in nisl porta scelerisque at id metus. Sed porttitor purus eu cursus porta. Donec suscipit semper ligula, ut eleifend nunc commodo ut. Aenean sed mauris molestie, gravida nisi et, lobortis sapien. Nullam et massa mi. </p> <p>Nulla a sem sit amet nisl interdum tincidunt. Fusce pellentesque nec risus et lacinia. Donec dignissim enim ac vehicula malesuada. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam at varius augue. Phasellus varius, sem ac posuere mattis, augue turpis sollicitudin arcu, ut tincidunt arcu tellus vel sapien. Nulla sagittis auctor velit, ut tristique nulla sollicitudin et. Duis ac purus efficitur, congue purus egestas, ultricies quam. Nulla tincidunt nibh risus, vel semper lacus consectetur quis. Aenean luctus enim in justo rhoncus dignissim. Donec pellentesque fermentum sapien vitae aliquam.</p><h2>Service #1</h2><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis nunc libero. Nullam tincidunt lectus ut maximus dignissim. Mauris et risus in metus ornare convallis. Nullam in orci leo. Aenean sit amet arcu vitae ipsum mollis blandit. Pellentesque pellentesque sapien mollis, pharetra magna sit amet, varius eros. Donec eu dui ullamcorper ligula rutrum tincidunt. Sed eu venenatis mi. Nullam rutrum sagittis tortor, id feugiat dolor commodo a. Nullam venenatis rutrum elementum. Vestibulum vehicula lectus in nisl porta scelerisque at id metus. Sed porttitor purus eu cursus porta. Donec suscipit semper ligula, ut eleifend nunc commodo ut. Aenean sed mauris molestie, gravida nisi et, lobortis sapien. Nullam et massa mi. </p> <p>Nulla a sem sit amet nisl interdum tincidunt. Fusce pellentesque nec risus et lacinia. Donec dignissim enim ac vehicula malesuada. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam at varius augue. Phasellus varius, sem ac posuere mattis, augue turpis sollicitudin arcu, ut tincidunt arcu tellus vel sapien. Nulla sagittis auctor velit, ut tristique nulla sollicitudin et. Duis ac purus efficitur, congue purus egestas, ultricies quam. Nulla tincidunt nibh risus, vel semper lacus consectetur quis. Aenean luctus enim in justo rhoncus dignissim. Donec pellentesque fermentum sapien vitae aliquam.</p>',
  },
  slug: 'services',
  meta: {
    metaDescription: 'string',
    metaKeywords: 'string',
    openGraph: {
      title: 'string',
      description: 'string',
      imageUrl: 'string',
    },
  },

  updatedAt: '2023-02-16T07:05:35.214Z',
};
