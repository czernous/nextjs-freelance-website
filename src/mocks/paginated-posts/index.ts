import { posts } from '../posts';

export const mockPagedData = {
  data: posts.slice(0, 10),
  hasPagination: true,
  page: 1,
  pageSize: 10,
  totalPages: Math.ceil(posts.length / 10),
  totalDocuments: posts.length,
};
