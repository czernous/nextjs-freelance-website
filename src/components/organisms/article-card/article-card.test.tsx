import { act, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import ArticleCard from './index';
import { cardMock, imageMock } from './mocks';
import { axe, toHaveNoViolations } from 'jest-axe';
import React from 'react';

expect.extend(toHaveNoViolations);

const mockedFetch = jest.fn();
global.fetch = mockedFetch;

jest.mock('../../../utils/data-fetching/client', () => ({
  fetchAndConvertToBase64: jest
    .fn()
    .mockImplementation(() => Promise.resolve('base64')),
}));

describe('Article Card', () => {
  let card;

  const mockResponse = { data: imageMock };

  mockedFetch.mockResolvedValueOnce({
    ok: true,
    json: async () => mockResponse,
  });
  beforeEach(async () => {
    await act(async () => {
      card = render(
        <ArticleCard
          imageUrl={cardMock.imageUrl}
          imageAltText={cardMock.imageAltText}
          blurredImageUrl={cardMock.blurredImageUrl}
          title={'Test Card'}
          description={'This is a test card'}
          ctaText={'Read More'}
          ctaUrl={'#'}
          unoptimized={false}
        />,
      );
    });
  });

  it('passes axe audit', async () => {
    const { container } = card;
    await act(async () => {
      expect(await axe(container)).toHaveNoViolations();
    });
  });

  it('is present in layout', () => {
    const card = screen.getByText('Test Card');
    waitFor(() => expect(card).toBeInTheDocument());
  });

  it('is rendered unchanged', () => {
    const { container } = card;
    expect(container).toMatchSnapshot();
  });
});
