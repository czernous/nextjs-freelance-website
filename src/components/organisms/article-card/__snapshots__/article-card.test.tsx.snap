// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Card is rendered unchanged 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="image"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="test image"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; filter: blur(20px); background-size: cover; background-image: url(/test-images/small.avif); background-position: 0% 0%;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="content"
    >
      <h3
        class="title"
      >
        Test Card
      </h3>
      <p
        class="description"
      >
        This is a test card
      </p>
      <a
        class="button brick regular  secondary  link "
        href="#"
      >
        Read More
      </a>
    </div>
  </div>
</div>
`;
